### ISSUE TITLE
RocksDB crashing when doing simple get

### ISSUE THREAD

**Issue by vsrivatsan89** [facebook/rocksdb#12362]
Using RocksDB version: `8.10.0`

```
    implementation 'org.rocksdb:rocksdbjni:8.10.0'
```

Initialized DB as such:
```
    RocksDB.loadLibrary();
    try (final Options options = new Options().setCreateIfMissing(true)) {
      try (final RocksDB db = RocksDB.open(options, "/mnt/test")) {
        log.info("Successfully loaded RocksDB");
        return db;
      } catch (RocksDBException e) {
        log.info("Exception in loading RocksDB");
        throw new RuntimeException(e);
      }
    }
```
Then I attempt the following code: 
``` try {
      byte[] value = db.get(db.getDefaultColumnFamily(), KEY);
      if (value == null) {
        value = ByteBuffer.allocate(4).putInt(1).array();
      } else {
        value = ByteBuffer.allocate(4).putInt(ByteBuffer.wrap(value).getInt() + 1).array();
      }
      db.put(db.getDefaultColumnFamily(), KEY, value);
      return "SUCCESS";
    } catch (RocksDBException e) {
      throw new RuntimeException(e);
    }
``` 

### Expected behavior
I would expect the code to fetch value if it exists, if not it puts it into the db with the associated key

### Actual behavior

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fc97dee1bf8, pid=1, tid=60
#
# JRE version: OpenJDK Runtime Environment Corretto-17.0.9.8.1 (17.0.9+8) (build 17.0.9+8-LTS)
# Java VM: OpenJDK 64-Bit Server VM Corretto-17.0.9.8.1 (17.0.9+8-LTS, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, linux-amd64)
# Problematic frame:
# C  [librocksdbjni14034371842470178076.so+0x2e1bf8]  rocksdb_get_helper(JNIEnv_*, rocksdb::DB*, rocksdb::ReadOptions const&, rocksdb::ColumnFamilyHandle*, _jbyteArray*, int, int)+0x128
#
# Core dump will be written. Default location: //core.1
#
# An error report file with more information is saved as:
# //hs_err_pid1.log
#
# If you would like to submit a bug report, please visit:
#   https://github.com/corretto/corretto-17/issues/
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#

[error occurred during error reporting (), id 0xb, SIGSEGV (0xb) at pc=0x00007fca4ec00898]
```

Checked the memory footprint and there is ~8GB of free memory at the time of crash

**Comment by adamretter** [comment]
If you want to explicitly use Column Families, you should pass the descriptors when you open the db, in return you will get back a set of handles that you can reuse for get/put.

Otherwise, if you don't want to explicitly use Column Families, you can just call - 

```java
byte[] value = db.get(KEY);

...

db.put(KEY, value);
```

**Comment by vsrivatsan89** [comment]
I thought that was the case, but I recall receiving exception saying that column family can’t be null when using the get (KEY) API. Let me try again.

**Comment by vsrivatsan89** [comment]
@adamretter I updated the code as per your suggestion to 
```
try {
      byte[] value = db.get(KEY);
      if (value == null) {
        value = ByteBuffer.allocate(4).putInt(1).array();
      } else {
        value = ByteBuffer.allocate(4).putInt(ByteBuffer.wrap(value).getInt() + 1).array();
      }
      db.put(KEY, value);
      return "SUCCESS";
    } catch (RocksDBException e) {
      throw new RuntimeException(e);
    }
```

I still get `SIGSEGV`. If i'm not using column families explicitly in my API calls, do I still need to register the default column family upon DB open?

**Comment by adamretter** [comment]
@vsrivatsan89 I just took a closer look at your earlier code:

```java
try (final Options options = new Options().setCreateIfMissing(true)) {
      try (final RocksDB db = RocksDB.open(options, "/mnt/test")) {
        log.info("Successfully loaded RocksDB");
        return db;
      } catch (RocksDBException e) {
        log.info("Exception in loading RocksDB");
        throw new RuntimeException(e);
      }
}
```

This is invalid! Your code will call `db.close()` and then `options.close()` before returning the `db` object.

**Comment by vsrivatsan89** [comment]
@adamretter Bah! Sorry about the silly mistake and wasting your time. Thank you for your patience.
